bokeh/plotting/gmap.py

```diff
@@ -229,6 +229,8 @@ def gmap(google_api_key, map_options, **kwargs):
     Returns:
        GMap
 
+    Note: The Google API key provided will be stored for the function to work.
```

Comment:
```suggestion
    The Google API key provided will be stored in the Bokeh Document JSON. 
```
Also I would suggest moving this sentence up above under the parameter info itself.